sound/soc/sof/intel/mtl.c

```diff
@@ -92,6 +92,13 @@ static int mtl_ipc_send_msg(struct snd_sof_dev *sdev, struct snd_sof_ipc_msg *ms
 {
 	struct sof_ipc4_msg *msg_data = msg->msg_data;
 
+	if (hda_ipc4_tx_is_busy(sdev)) {
+		sdev->ipc->sending_delayed = true;
```

Comment:
assuming the IPC times out even before we receive the ack from the firmware for the previous message, don't we also need to clear the sending_delayed flag somewhere?

Reply:
We are going to set it to false when a message is sent. If we time out, then at most we will have a new message to send, which we will try and based on the hardware state we will either delay it also or we are going to be able to send it.

On the other hand: There might be a case when we time out sending a message and then we receive a notification, which will trigger a re-send. Hrm. In that case the firmware is already in a strange state, but let I think we can handle that as well in upper layer clearing the flag after the completion handling.